compiler/luci/pass/src/FoldSparseToDensePass.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FoldSparseToDensePass.h"
+#include "CircleOptimizerUtils.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+/**
+ * Fold to const if
+ *
+ * 1. indices has 0-sized static shape (i.e., output is filled with default value)
```

Comment:
0-sized static shape means scalar?

Reply:
The second one. The word is confusing. Do you have a better word?

Reply:
maybe add more comments like `(for example, [0])` ?

Reply:
> maybe add more comments like (for example, [0]) ?

Added comments. PTAL

Reply:
I was confused because of `output is filled with default value`.
0-sized shape itself is okay for me but that sentence was little bit confused since 0-sized tensor cannot have a value.

Reply:
`output` in `output is filled with default value` and indices are different things?

Reply:
> output in output is filled with default value and indices are different things?

Yes. To understand the comment, you have to know the semantics of SparseToDense first.

SpareToDense has four inputs.

`indices`
`output_shape`
`values`
`default_value`

Output tensor is filled in two ways.
1. `indices` in output tensor is filled with `values`.
2. The rest of the tensor is filled with `default_value`.

So, if the size of `indices` is zero, the entire output tensor is filled with `default_value`.

Reply:
Now, I understand all clearly, thanks :)